compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -338,3 +375,59 @@ pub fn create_compressed_metadata_file(
 
     file.write().unwrap()
 }
+
+/// * Xcoff - AIX target does not support arbitrary section names, the metadata
+///   bytes are stored in the `.info` section that has the `STYP_INFO` flag.
+///   A global symbol is required to prevent the linker from throwing away the
+///   metadata section.
+pub fn create_rmeta_file_xcoff(Object: file, data: &[u8]) -> (Vec<u8>, MetadataPosition) {
+    assert!(file.format() == BinaryFormat::Xcoff);
+    file.add_section(
+        file.segment_name(StandardSegment::Text).to_vec(),
+        b".text".to_vec(),
+        SectionKind::Text,
+    );
+    let data_section = file.add_section(
+        file.segment_name(StandardSegment::Data).to_vec(),
+        b".data".to_vec(),
+        SectionKind::Data,
+    );
+    let section = file.add_section(
+        file.segment_name(StandardSegment::Debug).to_vec(),
+        b".info".to_vec(),
+        SectionKind::Debug,
+    );
+    file.add_file_symbol(".file".into());
+    file.section_mut(section).flags = SectionFlags::Xcoff { s_flags: (xcoff::STYP_INFO) as u32 };
+    // Add a symbol referring to .info section.
+    // Storage class will be handled in object crate.
+    file.add_symbol(Symbol {
+        name: "info".into(),
+        value: 4,
+        size: 0,
+        kind: object::SymbolKind::Null,
```

Comment:
Change to `SymbolKind::Unknown` and use `SymbolFlags::Xcoff` to set this to `C_INFO`.